airbyte-integrations/connectors/source-twilio/source_twilio/streams.py

```diff
@@ -90,45 +100,53 @@ def custom_transform_function(original_value: Any, field_schema: Mapping[str, An
                 # is no need in transforming anything.
                 pass
         return original_value
+        
 
 
-class IncrementalTwilioStream(TwilioStream, ABC):
+class IncrementalTwilioStream(TwilioStream, IncrementalMixin):
     cursor_field = "date_updated"
     time_filter_template = "%Y-%m-%dT%H:%M:%SZ"
-
-    def __init__(self, start_date: str = None, **kwargs):
+    
+    def __init__(self, start_date: str = None, lookback: int = 0, **kwargs):
         super().__init__(**kwargs)
-        self._start_date = start_date
+        self._start_date = start_date  
+        self._lookback = lookback
+        self._cursor_value = None
 
     @property
     @abstractmethod
     def incremental_filter_field(self) -> str:
         """
         return: date filter query parameter name
         """
+    
+    @property
+    def state(self) -> Mapping[str, Any]:
+        if self._cursor_value:
+            return  self._cursor_value
+        else:
+            return  self._start_date
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = value
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
-        """
-        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
-        and returning an updated state object.
-        """
-        latest_benchmark = pendulum.parse(latest_record[self.cursor_field], strict=False).strftime(self.time_filter_template)
-        if current_stream_state.get(self.cursor_field):
-            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
-        return {self.cursor_field: latest_benchmark}
 
     def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
         params = super().request_params(stream_state=stream_state, **kwargs)
-        start_date = stream_state.get(self.cursor_field) or self._start_date
+        start_date = stream_state or self.state
```

Comment:
```suggestion
        start_date = stream_state[self.cursor_field] if stream_state.get(self.cursor_field) else self.start_date
```

Reply:
You could define a small function to compute the start_date with the lookback window.

Reply:
```python
def compute_start_date(self, stream_state) -> str:
  start_date = stream_state[self.cursor_field] if stream_state.get(self.cursor_field) else self.start_date
  start_date = start_date - timedelta(minutes=self._lookback_window)
  return start_date.strftime(self.time_filter_template)
```
and call this function from `request_params`